### PR TITLE
chore(Dropdown): remove deprecated props

### LINKED ISSUE #8088: [Dropdown] - [remove deprecated props]
**Describe the problem**

Remove deprecated  props and any associated logic from following Dropdown components and subcomponents

- DropdownItem: `isHovered`
- DropdownMenu: `openedOnEnter`
- DropdownToggle: `isPrimary`
- Toggle: `isPrimary`

### PR DESCRIPTION
**Pull request by andyyvo**
**What**: Closes #8088 .

Removed all instances of:
- DropdownItem: `isHovered`
- DropdownMenu: `openedOnEnter`
- DropdownToggle: `isPrimary`
- Toggle: `isPrimary`

**Additional issues**:

### CODE REVIEW

**Comment by nicolethoen** [comment]
To do that, @andyyvo, just edit the PR title, and it'll give you a dropdown to choose which branch you'd like to merge your commits into. You should switch it from `patternfly:main` to `v5`

**Comment by andyyvo** [comment]
got it! thanks @jenny-s51 and @nicolethoen !!

**Review by jenny-s51 — APPROVED** [review]

File: packages/react-core/src/components/Dropdown/DropdownMenu.tsx [hunk 1]
```diff
@@ -180,7 +177,6 @@ export class DropdownMenu extends React.Component<DropdownMenuProps> {
       isGrouped,
       setMenuComponentRef,
       // eslint-disable-next-line @typescript-eslint/no-unused-vars
```
Comment (nicolethoen) [inline packages/react-core/src/components/Dropdown/DropdownMenu.tsx:179]: Is this comment still necessary?

Comment (andyyvo) [reply packages/react-core/src/components/Dropdown/DropdownMenu.tsx:179]: does this also apply to the comment on line 130 as well @nicolethoen ?

Comment (nicolethoen) [reply packages/react-core/src/components/Dropdown/DropdownMenu.tsx:179]: i does not

**Review by tlabaj — COMMENTED** [review]
LGTM

**Pull request merged**